Soru7.py

```diff
@@ -10,3 +10,9 @@
 # Kullanmanız gereken veri yapıları, bu veri yapilarinin nasil uygulamaya dokulecegi ve baskı hizmetini
 # etkin bir şekilde çalıştırmak için gerekli olacak algoritmalar hakkında yaklasik 10 cümle ile kısa bir paragraf yazın.
 # Seçimlerinizi çalışma süresi ve depolama boyutu açısından açıklayın. (Time & Space Complexity)
+
+
+# Burada FIFO mantigiyla bir algoritam yazmamiz gerekli. Cunku dosyasini ilk gonderenin cikitisinin ilk alinmasi lazim.
+# Bunu uygulamak icin bir stack uygulamasi gelistirmek lazim. Her ciktiya alinan dosyanin stackten silinmesi lazim.
+# Time complexity 1 cunku ilk elimize gelen dosyayi cagiriyoruz burda. Ama space complexity si fazla. Gonderilen datalar kadar yerimizin olmasi lazim.
+#
```

Comment:
Similar but a bit more comprehensive solution:

We will put in order the print requests by sending them to the end 
with enqueue() method, trying to print with dequeue() on the other hand 
by deleting in the queue.
For each printer, after each print operation, we can check whether 
the queue line(one printer) is empty by using isEmpty() and if it 
returns true, it means that we have a place to send the request. 
We may send the request to that printer.
In this case, we may think which printer would be efficient to 
send the request. We can determine the efficiency by storing the 
workloads(overflow and empty capacities) of printers in a hashtable 
and we can measure it by checking with one request. ( Queries in 
hastable takes O(1).)